amida-tech/qtc-web-ui
# Feature/QTC-159 Navigation Bar

# Navigation Bar for API React Frontend

(Wait on this PR until #85 and #87 are merged.)

## Description
### Related Tickets

- [QTC-159](https://jira.amida.com/browse/QTC-159)

### How Things Worked (or Didn't) Before This PR
We only had one React component, a frontend for the `mapping_data_frames` endpoint.

### How Things Work Now
The app uses React Router to create a navigation bar and display the current "page." React can only create single-page apps, so the router matches (artificial) URLs with different sets of HTML to make it seem like you're visiting separate pages. 

## How to Test

- Run `npm install`
- Run `npm start` to launch the app and wait for a few seconds until you see `｢wdm｣: Compiled successfully.`
- Open `localhost:1776` in your browser
- Click on the blue navigation links at the top to visit different "pages" on the app
   - See that the page's content changes
   - See that the URL changes (e.g. on the Create page, the URL becomes `http://localhost:1776/create`)
   - Create is the only one we've made so far, so Develop and Test simply return filler text.

## Readiness

1. [ ] This PR has full test coverage (If this PR fixes a bug, you _must_ add test cases respresentative of the bug).
2. [ ] This PR passes all automated tests.
3. [ ] This PR has no linting errors.
4. [ ] This PR has no hardcoded UI strings or other obvious issues.
5. [ ] In addition to testing my changes while running this code via `npm start`, I tested my changes while running this code on my machine via `npm run build-prod-start` (necessary because the latter command mimics production).
6. [ ] This PR's changes to configuration files have been documented in all appropriate places (such as but not limited to README.md), if applicable.
7. [ ] This PR's required changes outside of the scope of this repository have been documented in this pull request.
